### PR TITLE
How-To Sync Data : Add end 2 end test

### PR DESCRIPTION
**Pull request by Aschen**
### What's is in this PR ?

- [X] Change that adds new dependencies
- [X] New feature (non-breaking change which adds functionality)

### What's this PR do?

This PR add end2end test for the How-To Synchronize Kuzzle with another database.  
The tests ensure that the plugin and the scripts are still working.  
Cucumber is used for the tests with a ruby backend.  

### What are the relevant tickets?

 - https://github.com/kuzzleio/kuzzle-backlog/issues/250

### Where should the reviewer start?

 - https://github.com/kuzzleio/kuzzle-how-to/blob/add_test_for_sync_data/sync-data-to-another-database/features/step_definitions/sync_data.rb
 - https://github.com/kuzzleio/kuzzle-how-to/blob/add_test_for_sync_data/sync-data-to-another-database/features/sync_data.feature
 - https://github.com/kuzzleio/kuzzle-how-to/blob/add_test_for_sync_data/.travis.yml

### How should this be manually tested?

  - Step 1 : Go to `sync-data-to-another-database`
  - Step 2 : Run `./run_tests.sh`

### CODE REVIEW

File: sync-data-to-another-database/scripts/count_data.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
+const
+  Kuzzle = require('kuzzle-sdk'),
+  Cassandra = require('cassandra-driver'),
+  fs = require('fs');
+
+const cassandraClient = new Cassandra.Client({ contactPoints: [ 'cassandra', 'localhost' ] });
+const kuzzle = new Kuzzle('localhost', { port: 7512 })
+
+kuzzle
```
Comment (scottinet) [inline sync-data-to-another-database/scripts/count_data.js]: Consider flattening your promises calls, for instance:

```
kuzzle
  .collection('yellow-taxi', 'nyc-open-data')
  .countPromise({})
  .then(result => {
    console.log(`Kuzzle: Total documents in nyc-open-data.yellow-taxi : ${result}`)

    return cassandraClient.execute('SELECT COUNT(*) FROM nyc_open_data.yellow_taxi');
  })
  .then(result => {
    console.log(`Cassandra: Total lines in nyc_open_data.yellow_taxi : ${result.rows[].count}`);
    return kuzzle.disconnect();
  })
  .catch(error => {
    console.log(`Error: ${error}`)
    process.exit(1);
  });
```

Moreover, I think that force-exit with a status code of 0 is not clean. I would rather call `kuzzle.disconnect()` to close the network connection, that should be enough to empty the event loop, and at that point Node.js should exit by itself.

(you may have to do the same with the Cassandra client)

File: sync-data-to-another-database/plugin/lib/services/exporter.js [hunk 2]
```diff
@@ -25,18 +25,18 @@ class Exporter {
     return this.client
       .connect()
       .then(() => {
-        this.context.log.info('[kuzzle-plugin-export-cassandra] Cassandra client connected');
+        this.context.log.info('[kuzzle-plugin-sync-cassandra] Cassandra client connected');
         return Promise.resolve(true);
```
Comment (scottinet) [inline sync-data-to-another-database/plugin/lib/services/exporter.js]: In a promise chain, you don't have to create a promise to return an already resolved value.
Here, you may simply `return true;`

File: sync-data-to-another-database/plugin/lib/services/exporter.js [hunk 3]
```diff
@@ -25,18 +25,18 @@ class Exporter {
...
       })
       .catch(error => {
         if (this.config.retryCount <= 0) {
-          return Promise.reject(new this.context.errors.InternalError(`[kuzzle-plugin-export-cassandra] Unable to connect the client : ${error.message}`));
+          return Promise.reject(new this.context.errors.InternalError(`[kuzzle-plugin-sync-cassandra] Unable to connect the client : ${error.message}`));
```
Comment (scottinet) [inline sync-data-to-another-database/plugin/lib/services/exporter.js]: Nitpicking: you may use `throw` inside a promise chain to return a reject promise. This simplify the syntax a bit.

Moreover, I think that a [ExternalServiceError](https://docs.kuzzle.io/api-documentation/errors/#externalserviceerror) is more suited than a [InternalError](https://docs.kuzzle.io/api-documentation/errors/#internalerror):

```
throw new this.context.errors.ExternalServiceError(`[kuzzle-plugin-sync-cassandra] Unable to connect the client : ${error.message}`);
```

**Comment by scottinet** [comment]
Also, consider configuring and running `eslint` automatically, to enforce our syntax rules. You may directly copy the file `.eslintrc.json` at the root of the Kuzzle core project

File: sync-data-to-another-database/features/step_definitions/sync_data.rb [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+  end
+
+  if ! connected
+    puts docker_compose_stderr.read_nonblock(99999)
+    puts docker_compose_stdout.read_nonblock(99999)
+    raise ArgumentError, "Unable to start docker-compose stack"
+  end
+
+  puts "Kuzzle is up !"
+  sleep 5 # Wait for Cassandra initialization
```
[the remark this answers is not in the document]
Comment (Aschen) [reply sync-data-to-another-database/features/step_definitions/sync_data.rb]: I am only waiting to the keyspace initialisation, Cassandra is already up when Kuzzle is up

File: sync-data-to-another-database/features/step_definitions/sync_data.rb [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+
+    if status.exitstatus != 0
+      puts stderr.read
+      raise ArgumentError, "Fail to load dataset"
+    end
+  end
+end
+
+Then("I can check if they are synchronized in Cassandra") do
+  sleep 2 # Wait import to complete..
```
[the remark this answers is not in the document]
Comment (Aschen) [reply sync-data-to-another-database/features/step_definitions/sync_data.rb]: I still have to wait a little to let Cassandra refresh his indexes :/

**Pull request merged**